src/v/raft/log_eviction_stm.h

```diff
@@ -29,6 +30,10 @@ class log_eviction_stm {
 public:
     log_eviction_stm(consensus*, ss::logger&, ss::abort_source&);
 
+    void set_stm_manager(std::shared_ptr<storage::stm_manager> stm_manager) {
```

Comment:
`std::shared_ptr` doesn't work as expected in seastar. use one of the seastar variants which do not contain atomic counters.